tools/perf/ResultsComparer/src/ResultsComparer/Core/Reporting/MarkdownReporter.cs

```diff
@@ -0,0 +1,144 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MarkdownReporter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using MarkdownLog;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace ResultsComparer.Core.Reporting
+{
+    /// <summary>
+    /// Generates comparison reports in markdown format.
+    /// </summary>
+    public class MarkdownReporter : IReporter
+    {
+        /// <inheritdoc/>
+        public void GenerateReport(ComparerResults results, Stream destination, ComparerOptions args)
+        {
+            ComparerResult[] resultsArray = results.Results.ToArray();
+
+            if (resultsArray.Length == 0)
+            {
+                return;
+            }
+
+            using StreamWriter writer = new(destination, leaveOpen: true);

```

Comment:
I wonder whether there's a design we can adopt such that it's not up to the reporter to ensure that the stream is not closed. This feels too brittle

Reply:
Hmm, that's a good point. I feel like closing the stream by the caller is not a good idea because this method does not own the stream, and it may close the stream prematurely while the caller still intends to use it. What if I added a `leaveOpen` argument like `StreamWriter` that way there's flexibility, and the caller can instruct the reporter whether or not to leave the stream open, and the default could be to close the stream.